spec/models/g5_authenticatable/user_spec.rb

```diff
@@ -373,4 +375,116 @@
       end
     end
   end
+
+  describe '#update_roles_from_auth' do
+    before do
+      user.roles = []
+      user.save!
+    end
+
+    let(:user2) { G5Authenticatable::User.create(user_attributes) }
+    let(:user_attributes2) { FactoryGirl.attributes_for(:g5_authenticatable_user) }
+    let(:mock_urn) { 'mock_urn' }
+
+    class MockResource
+      attr_accessor :urn, :id
+      def initialize(urn)
+        @urn = urn
+        @id = 1
+      end
+      def self.where(params)
+        MockResourceList.new(params[:urn])
+      end
+    end
+    class MockResourceList
+      def initialize(urn)
+        @urn = urn
+      end
+      def first
+        MockResource.new(@urn)
+      end
+    end
```

Comment:
Instead of defining `MockResource` and `MockResourceList` as plain ruby objects, we can take advantage of rspec's own mocking framework. This has several advantages (such as making rspec automatically responsible for cleanup). 

```ruby
let(:mock_resource_class) { Class.new }
before { stub_const('MockResource', mock_resource_class) }

let(:mock_resource) { stub_model(mock_resource_class, urn: mock_urn) }
before do
  allow(mock_resource_class).to receive(:where).with(urn: mock_urn).and_return([mock_resource])
end
```

Note that the call to `stub_model` requires adding a dependency on [rspec-activemodel-mocks](https://github.com/rspec/rspec-activemodel-mocks) to the test group in the Gemfile. The `stub_model` method gives us a lot of convenience methods on the mock object, including an auto-generated ID. There are a couple tests here that are specifically checking for an ID of 1, so those would have to change to check for `mock_resource.id` instead.